interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,956 @@
+
+    if input_type == "included":
+        thermo_text.append("END\n\n\n")
+    else:
+        thermo_text.append("END\n")
+
+    return "".join(thermo_text)
+
+
+def write_transport_data(
+    species_list, filename="generated_transport.dat"
+):  # need to add notes
+    """Writes transport data to Chemkin-format file.
+    Parameters
+    ----------
+    species_list : list of cantera.Species
+        List of species objects
+    filename : path or str, optional
+        Filename for new Chemkin transport database file
+    """
+    geometry = {"atom": "0", "linear": "1", "nonlinear": "2"}
+    with open(filename, "w") as trans_file:
+
+        # write data for each species in the Solution object
+        for species in species_list:
+
+            # each line contains the species name, integer representing
+            # geometry, Lennard-Jones potential well depth in K,
+            # Lennard-Jones collision diameter in angstroms,
+            # dipole moment in Debye,
+            # polarizability in cubic angstroms, and
+            # rotational relaxation collision number at 298 K.
+            species_string = (
+                f"{species.name:<16}"
+                + f"{geometry[species.transport.geometry]:>4}"
+                + f"{(species.transport.well_depth / ct.boltzmann):>10.3f}"
+                + f"{(species.transport.diameter * 1e10):>10.3f}"
+                + f"{(species.transport.dipole / DEBYE_CONVERSION):>10.3f}"
+                + f"{(species.transport.polarizability * 1e30):>10.3f}"
+                + f"{species.transport.rotational_relaxation:>10.3f}"
+                + "\n"
+            )
+
+            trans_file.write(species_string)
+
+
+def write(
+    solution,
+    output_path=None,
+    reorder_reactions=False,
+    skip_thermo=False,
+    separate_thermo_file=False,
+    skip_transport=False,
+):
+    """Writes Cantera solution object to Chemkin-format file.
+    Parameters
+    ----------
+    solution : cantera.Solution
+        Model to be written
+    output_path : path or str, optional
+        Path of file to be written; if not provided, use cd / 'solution.name'
+    reorder_reactions : bool, optional
+        Flag to reorder reactions based upon molecular weight
+    skip_thermo : bool, optional
+        Flag to skip writing thermo data
+    separate_thermo_file : bool, optional
+        Flag to write thermo data in a seperate file from mechanism
+    skip_transport : bool, optional
+        Flag to skip writing transport data in separate file
+    Returns
+    -------
+    output_file_name : list
+        List of paths to output mechanism files (.ck, .therm, .tran)
+    Examples
+    --------
+    >>> gas = cantera.Solution('gri30.cti')
+    >>> yaml2ck.write(gas)
+    [output_path / reduced_gri30.ck, output_path / reduced_gri30.therm,
+     output_path / reduced_gri30.tran]
+    """
+    if output_path:
+        if not isinstance(output_path, PurePath):
```

Comment:
I may have gone a bit overboard on how I changed this, but I like the logic with how I've got it.

If a path is given that leads to a folder then it assigns solution name and appends the extension
If a path is given that is a file then it uses the cwd / file
If a str is given with just words then it uses cwd / file.{ext}

I'm sure it could be simplified but it works